tools/packaging/release/update-repository-version.sh

```diff
@@ -117,62 +150,60 @@ bump_repo() {
-		#  kata-deploy-base | NON-EXISTENT   | NON-EXISTENT   |
-		#  -----------------+----------------+----------------+
+		#                     +----------------+----------------+
+		#                     |      from      |       to       |
+		#  -------------------+----------------+----------------+
+		#  kata-deploy        | "x.y.z"        | "x.y.(z+1)"    |
+		#  -------------------+----------------+----------------+
+		#  kata-deploy-stable | NON-EXISTENT   | NON-EXISTENT   |
+		#  -------------------+----------------+----------------+
 
 		info "Updating kata-deploy / kata-cleanup image tags"
+		version_to_replace=${current_version}
 		if [ "${target_branch}" == "main" ] && [[ "${new_version}" =~ "rc" ]]; then
-			# case 2)
-			## change the "latest" tag to the "#{new_version}" one
-			sed -i "s#quay.io/kata-containers/kata-deploy:latest#quay.io/kata-containers/kata-deploy:${new_version}#g" tools/packaging/kata-deploy/kata-deploy/base/kata-deploy.yaml
-			sed -i "s#quay.io/kata-containers/kata-deploy:latest#quay.io/kata-containers/kata-deploy:${new_version}#g" tools/packaging/kata-deploy/kata-cleanup/base/kata-cleanup.yaml
-
-			git diff
-
-			git add tools/packaging/kata-deploy/kata-deploy/base/kata-deploy.yaml
-			git add tools/packaging/kata-deploy/kata-cleanup/base/kata-cleanup.yaml
-
-			## and remove the kata-deploy & kata-cleanup stable yaml files
+			## remove the kata-deploy & kata-cleanup stable yaml files
 			git rm tools/packaging/kata-deploy/kata-deploy/base/kata-deploy-stable.yaml
 			git rm tools/packaging/kata-deploy/kata-cleanup/base/kata-cleanup-stable.yaml
-		elif [[ "${target_branch}" =~ "stable" ]]; then
-			# case 3)
-			sed -i "s#quay.io/kata-containers/kata-deploy:${current_version}#quay.io/kata-containers/kata-deploy:${new_version}#g" tools/packaging/kata-deploy/kata-deploy/base/kata-deploy.yaml
-			sed -i "s#quay.io/kata-containers/kata-deploy:${current_version}#quay.io/kata-containers/kata-deploy:${new_version}#g" tools/packaging/kata-deploy/kata-cleanup/base/kata-cleanup.yaml
-			git diff
-
-			git add tools/packaging/kata-deploy/kata-deploy/base/kata-deploy.yaml
-			git add tools/packaging/kata-deploy/kata-cleanup/base/kata-cleanup.yaml
+
+			version_to_replace="latest"
 		fi
+
+		sed -i "s#quay.io/kata-containers/kata-deploy:${version_to_replace}#quay.io/kata-containers/kata-deploy:${new_version}#g" tools/packaging/kata-deploy/kata-deploy/base/kata-deploy.yaml
+		sed -i "s#quay.io/kata-containers/kata-deploy:${version_to_replace}#quay.io/kata-containers/kata-deploy:${new_version}#g" tools/packaging/kata-deploy/kata-cleanup/base/kata-cleanup.yaml
+
+		git diff
+
+		git add tools/packaging/kata-deploy/kata-deploy/base/kata-deploy.yaml
+		git add tools/packaging/kata-deploy/kata-cleanup/base/kata-cleanup.yaml
+
+		info "Creating the commit with the kata-deploy changes"
+		commit_msg="$(generate_kata_deploy_commit $new_version)"
+		git commit -s -m "${commit_msg}"
```

Comment:
- Isn't this missing `-a` or a list of files?
- Random idea: How about adding a `--dry-run` / `-n` option to this script so you can simulate the revert by getting the script to just show the commands it would run? When running in the normal mode, it would just call:

```bash
local cmds
cmds="git commit -s -m \"${commit_msg}\" ..."

if [ "$dry_run_mode" ]
then
  info "dry-run: Would run: $cmds"
else
    eval $cmds
fi
```

Or even better create a `run_commands()` function that either does the `info` or `eval` depending on `$dry_run_mode` to avoid littering the code with `if`/`else`/`fi` blocks. There is stealable code that does just that here:

- https://github.com/kata-containers/tests/blob/main/tracing/test-agent-shutdown.sh#L262..L274

Reply:
> Isn't this missing -a or a list of files?
Not really, a few lines above we do `git add ...` to the files we want to add and and `git rm ...` to the files we want to remove.

All other comments will be addressed as part of a v2.